bin/all_sky_search/pycbc_prepare_xml_for_gracedb

```
@@ -0,0 +1,230 @@
+    if args.event_id == coinc_event_id:
+        # This is the event we want:
+        break
+
+coinc_event_table_curr = lsctables.New(lsctables.CoincTable)
+coinc_event_table_curr.append(event)
+coinc_inspiral_table_curr = lsctables.New(lsctables.CoincInspiralTable)
+coinc_event_map_table_curr = lsctables.New(lsctables.CoincMapTable)
+sngl_inspiral_table_curr = lsctables.New(lsctables.SnglInspiralTable)
+
+for coinc_insp in coinc_inspiral_table:
+    if coinc_insp.coinc_event_id == event.coinc_event_id:
+        coinc_inspiral_table_curr.append(coinc_insp)
+
+time = coinc_inspiral_table_curr[0].end_time \
+        + coinc_inspiral_table_curr[0].end_time_ns * 1e-9
+
+sngl_ids = []
+for coinc_map in coinc_event_map_table:
+    if coinc_map.coinc_event_id == event.coinc_event_id:
+        coinc_event_map_table_curr.append(coinc_map)
+        sngl_ids.append(coinc_map.event_id)
+
+# Get the SNR timeseries and PSDs at the time of this event
+# these may not have contributed a trigger
+snr_timeseries_time_match = [snr_key for snr_key in snr_timeseries.keys()
+                             if abs(snr_key[1] - time) < 0.1]
+ifos = list(zip(*snr_timeseries_time_match))[0]
+time_key = list(zip(*snr_timeseries_time_match))[1][0]
+# These are the SNR timeseries for this event only:
+
+snr_curr_event = {}
+for ts_key in snr_timeseries_time_match:
+    snr_curr_event[ts_key[0]] = TimeSeries(
+        snr_timeseries[ts_key],
+        delta_t=snr_delta_t[ts_key[1]],
+        epoch=snr_epoch[ts_key[1]],
+        dtype=np.csingle,
+    )
+
+# IFOs from SNR timeseries:
+psds_curr_event = {}
+psddict = {}
+for ifo in ifos:
+    try:
+        psd = psds[ifo]
+    except KeyError:
+        parser.error(
+            "--psd-files {0}: no PSDs found for detector {1}".format(
+            " ".join(args.psd_files), ifo))
+
+    try:
+        psd = psd[psd.find(time)].psd
+    except ValueError:
+        logging.warning("SNR timeseries but no PSD exists for event at %.3f",
+            time)
```

What happens after the warning, everything just carries on? Is that what we want? Another case where I would just let the exception crash the code.